src/clarity-angular/layout/vertical-nav/vertical-nav.spec.ts

```diff
@@ -0,0 +1,630 @@
+                vertNavService.collapsible = true;
+                vertNavService.collapsed = true;
+
+                fixture.detectChanges();
+
+                expect(compiled.querySelector("clr-icon")).not.toBeNull();
+            });
+        });
+
+        describe("Nav Internals - Only Nav Groups", () => {
+            let vertNavService: VerticalNavService;
+
+            beforeEach(() => {
+                fixture = TestBed.createComponent(OnlyNavGroupTestComponent);
+                fixture.detectChanges();
+                compiled = fixture.nativeElement;
+                vertNavService = fixture.debugElement.query(By.directive(VerticalNav)).injector.get(VerticalNavService);
+            });
+
+            afterEach(() => {
+                fixture.destroy();
+            });
+
+            it("does not show .nav-group when nav is collapsed", () => {
+                vertNavService.collapsible = true;
+                vertNavService.collapsed = true;
+
+                fixture.detectChanges();
+
+                let displayStyle: string =
+                    window.getComputedStyle(<HTMLElement>compiled
+                        .querySelector(".nav-group"))
+                        .getPropertyValue("display");
+
+                expect(displayStyle).toBe("none");
+            });
+        });
+
+        describe("Nav Internals - Icons + Nav Groups", () => {
+            let vertNavService: VerticalNavService;
+
+            beforeEach(() => {
+                fixture = TestBed.createComponent(IconsAndNavGroupTestComponent);
+                fixture.detectChanges();
+                compiled = fixture.nativeElement;
+                vertNavService = fixture.debugElement.query(By.directive(VerticalNav)).injector.get(VerticalNavService);
+            });
+
+            afterEach(() => {
+                fixture.destroy();
+            });
+
+            it("shows the .nav-groups when the nav is expanded", () => {
+                expect(compiled.querySelector(".nav-group")).not.toBeNull();
+            });
+
+            it("shows the .nav-groups when the nav is collapsed", () => {
+                vertNavService.collapsible = true;
+                vertNavService.collapsed = true;
+
+                fixture.detectChanges();
+
+                expect(compiled.querySelector(".nav-group")).not.toBeNull();
+            });
+
+            it("clicking on .nav-group when nav is collapsed, expands the nav", () => {
+                vertNavService.collapsible = true;
+                vertNavService.collapsed = true;
+
+                fixture.detectChanges();
+
+                let navGroupTrigger: HTMLElement = <HTMLElement>compiled.querySelector(".nav-group-trigger");
+                navGroupTrigger.click();
+
+                fixture.detectChanges();
+
+                expect(vertNavService.collapsed).toBe(false);
+            });
+        });
+
+        describe("", () => {
```

Comment:
i didn't mean to keep this as empty. Just forgot to enter the description in a hurry.